crypto_kem/ledakemlt12/leaktime/H_Q_matrices_generation.c

```diff
@@ -1,32 +1,58 @@
 #include "H_Q_matrices_generation.h"
 #include "gf2x_arith_mod_xPplusOne.h"
 
-void PQCLEAN_LEDAKEMLT12_LEAKTIME_generateHPosOnes_HtrPosOnes(
-    POSITION_T HPosOnes[N0][DV],
-    POSITION_T HtrPosOnes[N0][DV],
-    AES_XOF_struct *keys_expander) {
+void PQCLEAN_LEDAKEMLT12_LEAKTIME_generateHPosOnes(POSITION_T HPosOnes[N0][DV],
+        AES_XOF_struct *keys_expander) {
+
     for (int i = 0; i < N0; i++) {
         /* Generate a random block of Htr */
-        PQCLEAN_LEDAKEMLT12_LEAKTIME_rand_circulant_sparse_block(&HtrPosOnes[i][0], DV, keys_expander);
-    }
-    for (int i = 0; i < N0; i++) {
-        /* Obtain directly the sparse representation of the block of H */
-        for (int k = 0; k < DV; k++) {
-            HPosOnes[i][k] = (P - HtrPosOnes[i][k])  % P; /* transposes indexes */
-        }
+        PQCLEAN_LEDAKEMLT12_LEAKTIME_rand_circulant_sparse_block(&HPosOnes[i][0],
+                DV,
+                keys_expander);
     }
 }
 
-void PQCLEAN_LEDAKEMLT12_LEAKTIME_generateQsparse(
-    POSITION_T pos_ones[N0][M],
-    AES_XOF_struct *keys_expander) {
+void PQCLEAN_LEDAKEMLT12_LEAKTIME_generateQPosOnes(POSITION_T QPosOnes[N0][M],
+        AES_XOF_struct *keys_expander) {
+
     for (int i = 0; i < N0; i++) {
         int placed_ones = 0;
         for (int j = 0; j < N0; j++) {
-            PQCLEAN_LEDAKEMLT12_LEAKTIME_rand_circulant_sparse_block(&pos_ones[i][placed_ones],
+            PQCLEAN_LEDAKEMLT12_LEAKTIME_rand_circulant_sparse_block(&QPosOnes[i][placed_ones],
                     qBlockWeights[i][j],
                     keys_expander);
             placed_ones += qBlockWeights[i][j];
         }
     }
 }
+
+void PQCLEAN_LEDAKEMLT12_LEAKTIME_transposeHPosOnes(POSITION_T HtrPosOnes[N0][DV], /* output*/
+        POSITION_T HPosOnes[N0][DV]) {
+
+    for (int i = 0; i < N0; i++) {
+        /* Obtain directly the sparse representation of the block of H */
+        for (int k = 0; k < DV; k++) {
+            HtrPosOnes[i][k] = (P - HPosOnes[i][k])  % P; /* transposes indexes */
+        }// end for k
+    }
+}
+
+void PQCLEAN_LEDAKEMLT12_LEAKTIME_transposeQPosOnes(POSITION_T QtrPosOnes[N0][M], /* output*/
+        POSITION_T QPosOnes[N0][M]) {
+
+    unsigned transposed_ones_idx[N0] = {0x00};
+    for (unsigned source_row_idx = 0; source_row_idx < N0 ; source_row_idx++) {
+        int currQoneIdx = 0; // position in the column of QtrPosOnes[][...]
+        int endQblockIdx = 0;
+        for (int blockIdx = 0; blockIdx < N0; blockIdx++) {
+            endQblockIdx += qBlockWeights[source_row_idx][blockIdx];
+            for (; currQoneIdx < endQblockIdx; currQoneIdx++) {
+                QtrPosOnes[blockIdx][transposed_ones_idx[blockIdx]] = (P -
```

Comment:
```suggestion
                QtrPosOnes[blockIdx][transposed_ones_idx[blockIdx]] =
```
move to line below (seriously annoying that these suggestions aren't multiline).